nautobot/docs/user-guide/administration/upgrading/from-v1/upgrading-from-nautobot-v1.md

```diff
@@ -30,6 +30,111 @@ $ nautobot-server pre_migrate
 All pre-migration checks passed.
 ```
 
+### Permission Constraint Migration
+
+Permission constraints that contain references to fields or models that were changed or removed in Nautobot 2.0 will have to be updated manually after the upgrade. For example, any permission constraints that reference a `Site` will need to be updated to reference the `Location` model instead. The `nautobot-server pre_migrate` command will output a list of all permission constraints that need to be updated after the upgrade.
+
+```no-highlight
+>>> Running permission constraint checks...
+
+One or more permission constraints may be affected by the Nautobot 2.0 migration.
+These permission constraints will have to be updated manually after upgrading to
+Nautobot 2.0 to reference new models and/or values. Please review the following
+warnings and make sure to document the objects referenced by these constraints
+before upgrading:
+
+ObjectPermission 'backbone devices' (id: ced686c3-2b34-4612-974a-bad766512661) has a constraint that references a model (nautobot.dcim.models.devices.DeviceRole) that will be migrated to a new model by the Nautobot 2.0 migration.
+{
+    "device_role": "e99adc77-40ef-4a0f-b2c1-26dbf6648ef1"
+}
+
+ObjectPermission 'example job run' (id: 41c6d03e-6388-47eb-b575-1c7a21725bc3) has a constraint that references a model field (nautobot.extras.models.jobs.Job.name) that may be changed by the Nautobot 2.0 migration.
+{
+    "name": "Example job, does nothing"
+}
+
+ObjectPermission 'emea' (id: 2d3b7aae-98ab-44ec-af89-43d3002a1b7d) has a constraint that references a model (nautobot.dcim.models.sites.Region) that will be migrated to a new model by the Nautobot 2.0 migration.
+[
+    {
+        "site__region__slug": "emea"
+    },
+    {
+        "id": "4c9f3e5c-2dc6-46f6-95ac-ac778369edfc"
+    }
+]
+```
+
+We recommend taking inventory of any objects referenced by primary key in permission constraints for the following models:
+
+- `dcim.DeviceRole`
+- `dcim.RackRole`
+- `extras.ObjectPermission`
+- `extras.TaggedItem`
+- `ipam.Aggregate`
+- `ipam.IPAddress`
+- `ipam.Prefix`
```

Comment:
Same ignorant question here as on #4445 - where do we expect ObjectPermission, IPAddress, and/or Prefix PKs to change in the migration?

Reply:
I removed ObjectPermission here but I think IPAddress/Prefix are good to leave in for now: https://github.com/nautobot/nautobot/pull/4445#discussion_r1327328629